plugins/common/opcua/input/input_client.go

```diff
@@ -425,7 +428,22 @@ func (o *OpcUAInputClient) MetricForNode(nodeIdx int) telegraf.Metric {
 		tags[k] = v
 	}
 
-	fields[nmm.Tag.FieldName] = o.LastReceivedData[nodeIdx].Value
+	// If the read UA node contains array elements, fan out the elements
+	// Otherwise creating a field with an array would fail, as the Type is not known to the convertField() function
+	if o.LastReceivedData[nodeIdx].ArrayLength > 0 {
+		valueArray := unpackArray(o.LastReceivedData[nodeIdx].Value)
+		if len(valueArray) == int(o.LastReceivedData[nodeIdx].ArrayLength) {
+			for index, value := range valueArray {
+				var sb strings.Builder
+				sb.WriteString(nmm.Tag.FieldName)
+				sb.WriteString(strconv.Itoa(int(index)))
+				fields[sb.String()] = value
+			}
+		}
```

Comment:
How about folding the `unpackArray`  function into here and reuse its `for` loop? This saves both allocations and loop iterations...